apis/fluent/atrium-api-fluent/src/commonTest/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/IterableLikeToContainInOrderOnlyGroupedCreatorSamples.kt

```diff
@@ -2,8 +2,239 @@ package ch.tutteli.atrium.api.fluent.en_GB.samples
+        expect(listOf("A", "B", "C", "D")).toContain.inOrder.only.grouped.within.inAnyOrder(
+            groupOf("A", "B"),        // first group: A and B (in any order)
+            groupOf("C", "D")         // second group: C and D (in any order)
+        )
+
+        // This also works - same groups, different internal order
+        expect(listOf("B", "A", "D", "C")).toContain.inOrder.only.grouped.within.inAnyOrder(
+            groupOf("A", "B"),        // first group: A and B (in any order)
+            groupOf("C", "D")         // second group: C and D (in any order)
+        )
+
+        // Single element groups
+        expect(listOf("A", "B", "C")).toContain.inOrder.only.grouped.within.inAnyOrder(
+            groupOf("A"),             // first group: single value A
+            groupOf("B", "C")         // second group: B and C (in any order)
+        )
+
+        fails { // because groups are in wrong order
+            expect(listOf("C", "D", "A", "B")).toContain.inOrder.only.grouped.within.inAnyOrder(
+                groupOf("A", "B"),    // first group should come first
+                groupOf("C", "D")     // second group should come second
+            )
+        }
+
+        fails { // because not all elements are present
+            expect(listOf("A", "B", "C")).toContain.inOrder.only.grouped.within.inAnyOrder(
+                groupOf("A", "B"),
+                groupOf("C", "D")     // D is missing from the list
+            )
+        }
+
+        fails { // because there are extra elements
+            expect(listOf("A", "B", "C", "D", "E")).toContain.inOrder.only.grouped.within.inAnyOrder(
+                groupOf("A", "B"),
+                groupOf("C", "D")     // E is not expected
+            )
+        }
+    }
+
+    @Test
+    fun inAnyOrderWithEntries() {
+        // Using expectation lambdas for more complex assertions
+        expect(listOf(1, 2, 10, 20)).toContain.inOrder.only.grouped.within.inAnyOrder(
+            groupOfEntries<Int>({ toBeGreaterThan(0) }, { toBeLessThan(5) }),      // first group: numbers > 0 and < 5
+            groupOfEntries<Int>({ toBeGreaterThan(5) }, { toBeGreaterThan(5) })   // second group: numbers > 5
+        )
+
+        // Mixed with null values
+        expect(listOf("A", null, "B", "C")).toContain.inOrder.only.grouped.within.inAnyOrder(
+            groupOfEntries<String>({ toEqual("A") }, null),                       // first group: "A" and null
+            groupOfEntries<String>({ toEqual("B") }, { toEqual("C") })           // second group: "B" and "C"
+        )
+
+        // Single entry in a group
+        expect(listOf(1, 10, 20)).toContain.inOrder.only.grouped.within.inAnyOrder(
+            groupOfEntries<Int>({ toBeLessThan(5) }),                             // first group: single entry < 5
+            groupOfEntries<Int>({ toBeGreaterThan(5) }, { toBeGreaterThan(5) })  // second group: numbers > 5
```

Comment:
```suggestion
            Entry { toBeLessThan(5) },                               // first group: single entry < 5
            Entries({ toBeGreaterThan(5) }, { toBeGreaterThan(5) })  // second group: numbers > 5
```

I hope `Int` is not required but I did not test it

Reply:
Int is indeed not required.